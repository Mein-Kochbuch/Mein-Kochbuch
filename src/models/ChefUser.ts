export type ChefUser = {
  id: string;
  name: string;
};
