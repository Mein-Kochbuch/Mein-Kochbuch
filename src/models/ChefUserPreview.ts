export type ChefUserPreview = {
  id: string;
  name: string;
};
