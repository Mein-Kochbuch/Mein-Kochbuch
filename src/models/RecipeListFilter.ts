export interface RecipeListFilter {
  sammlungsID?: number;
  count?: number;
  owner?: string;
  sortBy?: string;
  favorite?: boolean;
  maxDauer?: number;
  difficulty?: number;
  zutaten?: string[];
  searchWords?: string[];
}
