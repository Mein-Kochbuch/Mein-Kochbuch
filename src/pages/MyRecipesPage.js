import React from "react";
import Header from "../components/Header";
import {ScrollView, StyleSheet} from "react-native";
import MyRecipesList from "../components/MyRecipesList";
import useCookbooks from "../hooks/useCookbooks";

export default function MyRecipesPage() {
    const {cookbooks} = useCookbooks()

    return (
        <ScrollView style={styles.container}>
            <Header title={"My Recipes"}/>
            <MyRecipesList cookbooks={cookbooks}/>
        </ScrollView>
    );
}

const styles = StyleSheet.create({
    container: {
        height: "100%",
    },
})
