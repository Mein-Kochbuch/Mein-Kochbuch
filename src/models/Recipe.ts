import {Image} from './Image';
import {ChefUser} from './ChefUser';

export interface Recipe {
  id: string;
  title: string;
  owner: ChefUser;
  privacy: boolean;
  instruction: string;
  duration: number;
  difficulty: Difficulty;
  portions: number;
  thumbnail: Image;
  images: Image[];
  ingredients: Ingredient[];
  averageRating: number;
  ratingCount: number;
}

export type Ingredient = {
  id: string;
  amount: number;
  text: string;
};

export enum Difficulty {
  VERY_EASY = 0,
  EASY = 1,
  MEDIUM = 2,
  HARD = 3,
  EXPERT = 4,
}
