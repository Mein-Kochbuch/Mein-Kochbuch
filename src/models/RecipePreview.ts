import {Image} from './Image';
import {ChefUserPreview} from './ChefUserPreview';

export interface RecipePreview {
  id: string;
  title: string;
  averageRating: number;
  thumbnail?: Image;
  ratingCount: number;
  owner: ChefUserPreview;
}
