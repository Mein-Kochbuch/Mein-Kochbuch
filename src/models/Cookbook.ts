import {Image} from './Image';
import {Recipe} from './Recipe';
import {ChefUserPreview} from './ChefUserPreview';

export interface Cookbook {
  id: string;
  name: string;
  owner: ChefUserPreview;
  privacy: boolean;
  recipes: Recipe[];
  thumbnail?: Image;
}
