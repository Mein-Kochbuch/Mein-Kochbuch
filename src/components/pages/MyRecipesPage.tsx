import React from 'react';
import Header from './Header';
import {ScrollView, StyleSheet} from 'react-native';
import MyRecipesList from '../recipes/MyRecipesList';
import {Cookbook} from '../../models/Cookbook';

export type MyRecipesPageProps = {
  cookbooks: Cookbook[];
};

export default function MyRecipesPage({cookbooks}: MyRecipesPageProps) {
  return (
    <ScrollView style={styles.container}>
      <Header title={'My Recipes'} />
      <MyRecipesList cookbooks={cookbooks} />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    height: '100%',
  },
});
