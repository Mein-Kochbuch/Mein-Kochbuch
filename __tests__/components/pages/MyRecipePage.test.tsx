import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import MyRecipesPage from '../../../src/components/pages/MyRecipesPage';
import {Cookbook} from '../../../src/models/Cookbook';
import {Difficulty} from '../../../src/models/Recipe';

describe('MyRecipesPage', () => {
  it('Snapshot Test', () => {
    const cookbooks: Cookbook[] = [
      {
        recipes: [
          {
            title: 'recipe-1',
            id: '1',
            averageRating: 3.5,
            images: [],
            instruction: 'instructions-1',
            difficulty: Difficulty.EXPERT,
            portions: 4,
            privacy: false,
            duration: 15,
            ratingCount: 13,
            thumbnail: {
              id: '1',
              url: 'thumnail-url-1',
            },
            ingredients: [
              {
                id: '1',
                amount: 500,
                text: 'ingredient-1',
              },
            ],
            owner: {name: 'me', id: '1'},
          },
        ],
        id: '1',
        name: 'cookbook-name-1',
        privacy: false,
        owner: {
          name: 'me',
          id: '1',
        },
        thumbnail: {
          id: '1',
          url: 'thumbnail-url-1',
        },
      },
      {
        recipes: [
          {
            title: 'recipe-2',
            id: '2',
            averageRating: 3.0,
            images: [],
            instruction: 'instructions-2',
            difficulty: Difficulty.EASY,
            portions: 2,
            privacy: true,
            duration: 20,
            ratingCount: 1,
            thumbnail: {
              id: '2',
              url: 'thumnail-url-2',
            },
            ingredients: [
              {
                id: '2',
                amount: 500,
                text: 'ingredient-2',
              },
            ],
            owner: {name: 'you', id: '2'},
          },
        ],
        id: '2',
        name: 'cookbook-name-2',
        privacy: false,
        owner: {
          name: 'you',
          id: '2',
        },
        thumbnail: {
          id: '2',
          url: 'thumbnail-url-2',
        },
      },
    ];

    const component = renderer.create(<MyRecipesPage cookbooks={cookbooks} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
