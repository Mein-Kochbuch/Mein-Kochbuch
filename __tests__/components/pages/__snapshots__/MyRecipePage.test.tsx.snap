// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyRecipesPage Snapshot Test 1`] = `
<RCTScrollView
  style={
    Object {
      "height": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "display": "flex",
          "justifyContent": "center",
          "marginVertical": 12,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 50,
            "textAlign": "center",
          }
        }
      >
        My Recipes
      </Text>
    </View>
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../resources/platzhalter.png",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#68a0cf",
                "borderColor": "#fff",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "paddingBottom": 20,
                "paddingTop": 20,
                "width": 100,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "marginLeft": 12,
                "textAlign": "center",
              },
            ]
          }
        >
          All Recipes
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "opacity": 1,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 10,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#FF0000",
                "fontSize": 100,
              },
              undefined,
              Object {
                "fontFamily": "Feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ÔÜÅ
        </Text>
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "marginLeft": 12,
                "textAlign": "center",
              },
            ]
          }
        >
          Favorites
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "marginBottom": 12,
              "marginLeft": 12,
              "marginRight": 12,
              "marginTop": 12,
            },
          ]
        }
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "height": 120,
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "thumbnail-url-1",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#68a0cf",
                "borderColor": "#fff",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "paddingBottom": 20,
                "paddingTop": 20,
                "width": 100,
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 30,
                "marginLeft": 12,
                "maxWidth": "70%",
                "textAlign": "center",
              },
            ]
          }
        >
          cookbook-name-1
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "#222",
            "borderRadius": 10,
            "borderStyle": "solid",
            "borderWidth": 2,
            "display": "flex",
            "flexDirection": "row",
            "height": 120,
            "marginBottom": 12,
            "marginLeft": 12,
            "marginRight": 12,
            "marginTop": 12,
            "opacity": 1,
            "paddingBottom": 12,
            "paddingLeft": 12,
            "paddingRight": 12,
            "paddingTop": 12,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "thumbnail-url-2",
            }
          }
          style={
            Array [
              Object {
                "backgroundColor": "#68a0cf",
                "borderColor": "#fff",
                "borderRadius": 10,
                "borderWidth": 1,
                "height": 100,
                "paddingBottom": 20,
                "paddingTop": 20,
                "width": 100,
              },
            ]
          }
        />
        <Text
          ellipsizeMode="middle"
          numberOfLines={2}
          style={
            Array [
              Object {
                "fontSize": 30,
                "marginLeft": 12,
                "maxWidth": "70%",
                "textAlign": "center",
              },
            ]
          }
        >
          cookbook-name-2
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
